### PR TITLE
Add org.openhantek.OpenHantek6022

### PR DESCRIPTION
**Pull request by kobus-v-schoor**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
`device=all` is needed for the software to be useful (frontend for USB measurement device)
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/OpenHantek/OpenHantek6022/issues/170
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by kobus-v-schoor** [comment]
Hi, this package requires a udev rule to be added to the host system to work properly, I don't know if this is currently possible with Flatpak? If not, would it be possible to add the step to the description of the package's Flathub page? Or is there a better way that it is usually handled?

**Comment by LeandroStanger** [comment]
bot, build org.openhantek.OpenHantek6022

File: org.openhantek.OpenHantek6022.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+app-id: org.openhantek.OpenHantek6022
+# for qt libs
+runtime: org.kde.Platform
+runtime-version: "5.15"
+sdk: org.kde.Sdk
+command: OpenHantek
+finish-args:
+  # allow displaying on x11 and wayland
+  - --socket=x11
```
Comment (hfiguiere) [inline org.openhantek.OpenHantek6022.yml:9]: x11 need `--share=ipc`

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
Build fail because the appdata is missing.

**Comment by kobus-v-schoor** [comment]
bot, build org.openhantek.OpenHantek6022

**Comment by kobus-v-schoor** [comment]
Just tested the build locally and everything was working as expected. This build does however still require manually adding udev rules (I've added a note about this in the appdata description) - just want to check if there is perhaps an alternative that allows automatically adding the udev rules?

**Comment by kobus-v-schoor** [comment]
I've found the following references to the udev rules problem:

- https://github.com/flathub/com.valvesoftware.Steam/issues/8
- https://github.com/flatpak/flatpak/issues/961
- https://github.com/Nitrokey/nitrokey-app/issues/435

It seems that adding the udev rules automatically is not an option. Luckily the app already has a warning at startup indicating that the udev rules need to be installed and the appdata description now also mentions this so I think it isn't too much of a problem.

**Comment by nedrichards** [comment]
Yeah, if you've got it in both areas then that's acceptable although obviously it's not an ideal experience for anyone.

**Pull request closed without merging**